src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int BIT_DIVIDER = 1;
+    private static final int SOURCE_POSITION = 0;
+    private static final int DESTINATION_POSITION = 0;
+    private static final int SHIFT_INDEX_NUMBER = 1;
+    private static final int SIZE_INITIALIZER = 0;
+    private int size;
+    private Object[] elements;
+
+    public ArrayList() {
+        size = SIZE_INITIALIZER;
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayIsFull();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index, index > size());
+        growIfArrayIsFull();
+        Object[] elements = this.elements;
+        System.arraycopy(elements, index, elements, index + SHIFT_INDEX_NUMBER, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        growForAddList(list.size());
+        for (int i = 0; i < list.size(); i++) {
+            elements[size + i] = list.get(i);
+        }
+        size += list.size();
```

Comment:
use the `add` method
```suggestion
        for (int i = 0; i < list.size(); i++) {
            add(list.get(i));
        }
```